File: Week1/problem_set/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)

{
    // Get change from user
    float dollars;

    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    // Convert dollars to cents
    int cents = round(dollars * 100);

    // Calculate minimum number of coins required
    int amount_of_coins = cents / 25;
    cents = cents % 25;
    amount_of_coins = amount_of_coins + cents / 10;
    cents = cents % 10;
    amount_of_coins = amount_of_coins + cents / 5;
    cents = cents % 5;
    amount_of_coins = amount_of_coins + cents;

    // Print the result
    printf("%i\n", amount_of_coins);
}

File: Week6/problem_set/cash/cash.py
from cs50 import get_float


def main():
    while True:
        change_owed_in_dollars = get_float("Change owed: ")
        if change_owed_in_dollars >= 0:
            break
    
    change_owed_in_cents = int(change_owed_in_dollars * 100)
    amount_of_coins = calculate_amount_of_coins(change_owed_in_cents)
    print(amount_of_coins)


def calculate_amount_of_coins(change_owed):
    amount_of_coins = int(change_owed / 25)
    amount_of_coins += int(change_owed % 25 / 10)
    amount_of_coins += int(change_owed % 25 % 10 / 5)
    amount_of_coins += int(change_owed % 25 % 10 % 5 / 1)
    
    return amount_of_coins
    

if __name__ == "__main__":
    main()